src/Uno.UI/Microsoft/UI/Xaml/Controls/PullToRefresh/RefreshContainer/Native/NativeRefreshControl.Android.cs

```diff
@@ -0,0 +1,293 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+using Android.Views;
+using Android.Widget;
+using AndroidX.Core.View;
+using AndroidX.RecyclerView.Widget;
+using AndroidX.SwipeRefreshLayout.Widget;
+using Uno.UI.Controls;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+
+namespace Uno.UI.Xaml.Controls;
+
+public partial class NativeRefreshControl : SwipeRefreshLayout, IShadowChildrenProvider, DependencyObject, ILayouterElement
+{
+	// Distance in pixels a touch can wander before we think the user is scrolling
+	// https://developer.android.com/reference/android/view/ViewConfiguration.html#getScaledTouchSlop()
+	private static readonly float _touchSlop = ViewConfiguration.Get(ContextHelper.Current).ScaledTouchSlop;
+	private System.Drawing.PointF _gestureStart;
+	private bool _isSwiping = false;
+	private bool _ignoreGesture = false;
+	private Android.Views.View _content;
+	private ViewGroup _descendantScrollable;
+	private bool _lastOnInterceptTouchEvent;
+
+	public NativeRefreshControl() : base(ContextHelper.Current)
+	{
+		_layouter = new NativeRefreshControlLayouter(this);
+	}
+
+	internal Android.Views.View Content
+	{
+		get => _content;
+		set
+		{
+			if (_content != null)
+			{
+				RemoveView(_content);
+			}
+			_content = value;
+			if (_content != null)
+			{
+				AddView(_content);
+			}
+		}
+	}
+
+	public override bool CanChildScrollUp()
+	{
+		_descendantScrollable = _descendantScrollable ?? GetDescendantScrollable();
+
+		if (_descendantScrollable != null)
+		{
+#pragma warning disable CS0618 // Type or member is obsolete
+			var canScrollUp = ViewCompat.CanScrollVertically(_descendantScrollable, -1);
+#pragma warning restore CS0618 // Type or member is obsolete
+			return canScrollUp;
+		}
+		return base.CanChildScrollUp();
+	}
+
+	private readonly List<View> _emptyList = new();
+	//public override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec) => base.OnMeasure(widthMeasureSpec, heightMeasureSpec);
+	List<View> IShadowChildrenProvider.ChildrenShadow => Content != null ? new List<View>(1) { Content as View } : _emptyList;
+
+	private ILayouter _layouter;
+
+	ILayouter ILayouterElement.Layouter => _layouter;
+	Size ILayouterElement.LastAvailableSize => LayoutInformation.GetAvailableSize(this);
+	bool ILayouterElement.IsMeasureDirty => true;
```

Comment:
I kinda stole this layouting code from `ScrollContentPresenter` 😁 . But it was needed, as the `SwipeRefreshLayout` internally was incorrectly measuring the content and giving it zero height for measure instead of positive infinity. Ideally in next iteration I will just make a full custom version of `SwipeRefreshLayout` including the customization like own visualizer.